### DISCUSSION TITLE
How to save/load model

### DISCUSSION THREAD

**Discussion by yiqiang-zhao** [hpcaitech/ColossalAI#1870] [Community | Q&A]
Hi, I want to save a model and load it using `torch.load`. I followed [this](https://github.com/hpcaitech/ColossalAI-Examples/blob/main/language/bert/colotensor/train.py) and add a `SaveCheckpointHook` in the `hook_list`.

I can load the checkpointed model, but the shape of every parameter is reduced to 1/8 by one dimension. Besides, every parameter's name is prefixed with `module.module.`.

I used a 8-v100-16G node.

Are there any tutorials on saving model and loading it by pure torch? Thanks.

**Comment by ver217** [comment]
What parallel training technique are you using?

**Reply by Excuses123** [reply]
Are you using tensor parallel? I also encounter this problem

**Reply by yangjianxin1** [reply]
I use tensor parallel, and  I also encounter this problem

**Reply by yangjianxin1** [reply]
do you solve the problem @Excuses123 @ver217

**Reply by binmakeswell** [reply]
Hi @yiqiang-zhao @Excuses123 @yangjianxin1 We will update this part next week. https://github.com/hpcaitech/ColossalAI/discussions/3339 Thanks.
